subprojects/core/src/main/java/org/gradle/internal/service/scopes/VirtualFileSystemServices.java

```diff
@@ -411,8 +412,13 @@ OutputSnapshotter createOutputSnapshotter(FileCollectionSnapshotter fileCollecti
             return new DefaultOutputSnapshotter(fileCollectionSnapshotter);
         }
 
-        AbsolutePathFileCollectionFingerprinter createAbsolutePathFileCollectionFingerprinter(FileCollectionSnapshotter fileCollectionSnapshotter) {
-            return new AbsolutePathFileCollectionFingerprinter(DirectorySensitivity.DEFAULT, fileCollectionSnapshotter);
+        Supplier<AbsolutePathFileCollectionFingerprinter> createDefaultAbsoluteFileCollectionFingerprinterSupplier(FileCollectionSnapshotter fileCollectionSnapshotter) {
+            AbsolutePathFileCollectionFingerprinter defaultAbsolutePathFileCollectionFingerprinter = new AbsolutePathFileCollectionFingerprinter(DirectorySensitivity.DEFAULT, fileCollectionSnapshotter);
+            return () -> defaultAbsolutePathFileCollectionFingerprinter;
+        }
```

Comment:
I don't think this is making things better, let's revert this and pass the registry for now. We can revisit this later I think.

Reply:
Agreed.  Done.